### PR TITLE
Add tier 4 unit tests (bulk_loader, interfaces, connection)

### PR DESCRIPTION
**Pull request by berendt**
Implements #546 (tier 4 of the meta test-coverage effort #524).

Adds unit tests for the **bulk loader & connection** modules under
`files/netbox/` — `bulk_loader.py`, `interfaces.py` and `connection.py`.
This is the first tier where the code under test wraps the pynetbox
session itself, so the tests move from `SimpleNamespace`-only fakes to
`MagicMock`-backed sessions where call counts, call arguments and
per-call `side_effect` failures matter. No live NetBox is required.

All three modules reach **100% line coverage** and the tier-4 suite runs
in well under a second with no real waiting.

## Change set (commit order)

1. **Extend conftest factories for tier 4 modules** — extends the
   `tests/unit/netbox/conftest.py` factories introduced in #543, all
   backward-compatibly (keyword-only, defaulted), so the tier-3 tests
   keep passing unchanged:
   - `make_interface` gains `name`, `mac_address`, `untagged_vlan`,
     `device` (and `mgmt_only` now defaults to `False`)
   - `make_ip` gains a keyword-only `assigned_object_id`
   - new `make_vlan(vid)` helper

2. **Add tier 4 unit tests for the bulk loader** — `test_bulk_loader.py`
   drives `BulkDataLoader` with a `MagicMock` pynetbox session:
   constructor defaults / `BATCH_SIZE` parity, `_chunk_list` edge cases,
   `load_device_data` (empty short-circuit, single/multi-batch interface
   loading asserted via `call_args_list`, IP batching + grouping, partial
   per-batch failures that `continue`, the no-interfaces skip, the
   grouping `AttributeError` that clears state and re-raises, additive
   accumulation across calls), the cache getters, `clear()` and
   `get_statistics()`.

3. **Add tier 4 unit tests for the interface handler** —
   `test_interfaces.py` uses the *real* in-memory `CacheManager` and the
   `make_fake_api` stub: `get_oob_interface` cache hit / ideal IP+MAC
   match / cached-on-second-call / IP-preferred-over-MAC-only /
   MAC-only fallback / first-fallback-wins / skip rules / cached negative
   result / VLAN extraction / filter-exception-without-caching, plus the
   `_is_managed_oob_interface` truth table (including the no-`hasattr`
   asymmetry vs. the gnmic twin), `_get_vlan_id` and `_get_interface_ip`.

4. **Add tier 4 unit tests for the connection manager** —
   `test_connection.py` monkeypatches `connection.pynetbox.api`,
   `connection.time.sleep` and the `connection.requests` SSL helpers in
   the module-under-test namespace; `time.sleep` is patched in every
   `connect()` test. Covers `__init__`, `connect` (first-attempt success,
   the SSL-ignore branch, the no-SSL branch, retry-then-success,
   retry exhaustion raising `NetBoxConnectionError` chained from the
   original cause, and `pynetbox.api` itself raising),
   `_configure_ssl_ignore` (with and without `self.api`), and
   `disconnect` (close, idempotency, no-prior-connect).

## Verification

- `pytest tests/unit` — 325 passed
- `pytest --cov=bulk_loader --cov=interfaces --cov=connection` — 100% on
  all three modules
- `pytest --durations=10` — tier-4 suite 0.17s, all durations <0.005s
- `flake8 files tests` — clean
- `black --check` on the four touched files — clean

## Notes

- Coverage is invoked as `--cov=bulk_loader` (etc.) rather than the
  issue's `--cov=files.netbox.bulk_loader`: the test harness puts
  `files/netbox/` on `sys.path` (see `tests/conftest.py`) so the modules
  import flat, and the flat name is what coverage matches. The reported
  file is still `files/netbox/bulk_loader.py`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/unit/netbox/test_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        assert handler.cache.get("oob_interface_1") is None
+        # ...and a subsequent call queries the API again.
+        handler.get_oob_interface(device)
+        assert api.dcim.interfaces.filter.call_count == 2
+
+
+# ---------------------------------------------------------------------------
+# _is_managed_oob_interface
+#
+# Unlike the gnmic_extractor twin (#543), this reads interface.tags directly
```
Comment (ideaship) [inline tests/unit/netbox/test_interfaces.py]: This contrast is inaccurate. `files/netbox/interfaces.py:115` and the gnmic twin `files/netbox/extractors/gnmic_extractor.py:149` both read `interface.tags` directly with the same `if not interface.tags or not interface.mgmt_only:` guard — neither has a `hasattr` guard at the interface level. The only `hasattr` guard in gnmic is on `device.tags` (`gnmic_extractor.py:74`), a different method on a different object. Please drop the "Unlike the gnmic_extractor twin … no hasattr guard" clause; the two paths handle tags identically. The "stubs must always define tags" note is fine to keep.

_Classification: cleanup only._

File: tests/unit/netbox/test_connection.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        assert manager.api.http_session.verify is False
+
+    def test_no_ssl_branch_leaves_session_untouched(self, patched, disable_warnings):
+        patched.fake_api.dcim.sites.count.return_value = 5
+        manager = ConnectionManager(_config(ignore_ssl_errors=False))
+        manager.connect()
+        disable_warnings.assert_not_called()
+        assert manager._session is None
+
+    def test_retry_then_success_sleeps_once(self, patched):
```
Comment (ideaship) [inline tests/unit/netbox/test_connection.py:103]: The retry tests here all run with the default `ignore_ssl_errors=False`, and the only SSL-on test (`test_ignore_ssl_errors_configures_session`, line 88) succeeds on the first attempt — so the SSL-ignore × retry intersection is never exercised. In production `_configure_ssl_ignore()` runs inside the retry loop on every attempt (`files/netbox/connection.py:41-42`) and rebinds `self._session = requests.Session()` (`connection.py:61`) without closing the previous session. So with `ignore_ssl_errors=True` and a failing `dcim.sites.count()` probe, each retry leaks the prior `requests.Session` (and its connection pool) until GC reclaims it.

Worth a test that drives ≥2 attempts with `ignore_ssl_errors=True` to characterize this, and ideally a follow-up that closes the old `_session` before rebinding in `connection.py`. Low severity — no wrong result, and it only triggers on the SSL-ignore-and-failing-probe path.

_Classification: missing/weak test (underlying pre-existing production concern: resource leak)._

**Review by ideaship — CHANGES_REQUESTED** [review]
Minor issues.

**Review by ideaship — APPROVED** [review]
